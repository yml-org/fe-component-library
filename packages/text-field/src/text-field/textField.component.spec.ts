import { LitElement } from 'lit-element';

describe('text-field', () => {
  const TEXT_FIELD = 'text-field';
  let textFieldElement: LitElement;

  const getShadowRoot = (tagName: string): ShadowRoot =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    textFieldElement = window.document.createElement(TEXT_FIELD) as LitElement;
    document.body.appendChild(textFieldElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(TEXT_FIELD)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(document.body.getElementsByTagName(TEXT_FIELD)[0]).toMatchSnapshot();
  });

  it('matches the label passed', async () => {
    const LABEL = 'Label';
    textFieldElement['label'] = LABEL;
    await textFieldElement.updateComplete;
    const textEle = getShadowRoot(TEXT_FIELD).querySelector('label').innerText;
    console.log(textEle);
    expect(textEle).toEqual(LABEL);
  });

  it('renders disabled text field', async () => {
    textFieldElement['disabled'] = true;
    await textFieldElement.updateComplete;
    const isDisabled =
      getShadowRoot(TEXT_FIELD).querySelector('input').disabled;
    expect(isDisabled).toBeTruthy();
  });

  it('focus on the text field if autoFocus is true', async () => {
    textFieldElement['autofocus'] = true;
    await textFieldElement.updateComplete;
    const isAutoFocus =
      getShadowRoot(TEXT_FIELD).querySelector('input').autofocus;
    expect(isAutoFocus).toBeTruthy();
  });

  it('renders readonly text field', async () => {
    textFieldElement['readonly'] = true;
    await textFieldElement.updateComplete;
    const isReadOnly =
      getShadowRoot(TEXT_FIELD).querySelector('input').readOnly;
    expect(isReadOnly).toBeTruthy();
  });

  it('has error if error prop is passed', async () => {
    textFieldElement['error'] = true;
    await textFieldElement.updateComplete;
    const textFieldComponentClassList =
      getShadowRoot(TEXT_FIELD).querySelector('div').classList;
    expect(textFieldComponentClassList).toContain('text-field-wrapper-error');
  });

  it('renders a required text field', async () => {
    textFieldElement['required'] = true;
    await textFieldElement.updateComplete;
    const isRequired =
      getShadowRoot(TEXT_FIELD).querySelector('input').required;
    expect(isRequired).toBeTruthy();
  });
});
