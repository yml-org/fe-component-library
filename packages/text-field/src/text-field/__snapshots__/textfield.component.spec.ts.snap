// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-field matches component snapshot 1`] = `
u {
  "_$AL": Map {},
  "_$Do": N {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": mt {
      "_$AD": H {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "class",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 1,
            "type": 2,
          },
          Object {
            "ctor": [Function],
            "index": 3,
            "name": "click",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "class",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "id",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "value",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "placeholder",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "autofocus",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "disabled",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "readonly",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "required",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "pattern",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "minlength",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "maxlength",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 5,
            "name": "input",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 6,
            "type": 2,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class=""
            part=""
          >
            
      
            <!--?lit$569777623$-->
            
      
            <div
              class="text-field-input-wrapper"
            >
              
        
              <div
                part="text-field-icon"
              >
                
          
                <slot
                  name="textfield-icon"
                />
                
        
              </div>
              
        
              <input
                autofocus="false"
                class="text-field-input"
                id=""
                part="text-field-input"
                placeholder=""
                type="text"
                value=""
              />
              
      
            </div>
            
      
            <!--?lit$569777623$-->
            
    
          </div>,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class=""
            part=""
          >
            
      
            <!--?lit$569777623$-->
            
      
            <div
              class="text-field-input-wrapper"
            >
              
        
              <div
                part="text-field-icon"
              >
                
          
                <slot
                  name="textfield-icon"
                />
                
        
              </div>
              
        
              <input
                autofocus="false"
                class="text-field-input"
                id=""
                part="text-field-input"
                placeholder=""
                type="text"
                value=""
              />
              
      
            </div>
            
      
            <!--?lit$569777623$-->
            
    
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        N {
          "_$AA": <!--?lit$569777623$-->,
          "_$AB": 
      ,
          "_$AH": Symbol(lit-nothing),
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        _t {
          "_$AH": [Function],
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            part="text-field-icon"
          >
            
          
            <slot
              name="textfield-icon"
            />
            
        
          </div>,
          "name": "click",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 5,
        },
        R {
          "_$AH": "text-field-input",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "class",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "id",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "value",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "placeholder",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": false,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "autofocus",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        bt {
          "_$AH": false,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "disabled",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        bt {
          "_$AH": false,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "readonly",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        bt {
          "_$AH": false,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "required",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        bt {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "pattern",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        bt {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "minlength",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        bt {
          "_$AH": "",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "maxlength",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 4,
        },
        _t {
          "_$AH": [Function],
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <input
            autofocus="false"
            class="text-field-input"
            id=""
            part="text-field-input"
            placeholder=""
            type="text"
            value=""
          />,
          "name": "input",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 5,
        },
        N {
          "_$AA": <!--?lit$569777623$-->,
          "_$AB": 
    ,
          "_$AH": Symbol(lit-nothing),
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__changeHandler": "",
  "__clickIconHandler": "",
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": N {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": mt {
        "_$AD": H {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "class",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 1,
              "type": 2,
            },
            Object {
              "ctor": [Function],
              "index": 3,
              "name": "click",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "class",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "id",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "value",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "placeholder",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "autofocus",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "disabled",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "readonly",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "required",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "pattern",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "minlength",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "maxlength",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 5,
              "name": "input",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 6,
              "type": 2,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class=""
              part=""
            >
              
      
              <!--?lit$569777623$-->
              
      
              <div
                class="text-field-input-wrapper"
              >
                
        
                <div
                  part="text-field-icon"
                >
                  
          
                  <slot
                    name="textfield-icon"
                  />
                  
        
                </div>
                
        
                <input
                  autofocus="false"
                  class="text-field-input"
                  id=""
                  part="text-field-input"
                  placeholder=""
                  type="text"
                  value=""
                />
                
      
              </div>
              
      
              <!--?lit$569777623$-->
              
    
            </div>,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class=""
              part=""
            >
              
      
              <!--?lit$569777623$-->
              
      
              <div
                class="text-field-input-wrapper"
              >
                
        
                <div
                  part="text-field-icon"
                >
                  
          
                  <slot
                    name="textfield-icon"
                  />
                  
        
                </div>
                
        
                <input
                  autofocus="false"
                  class="text-field-input"
                  id=""
                  part="text-field-input"
                  placeholder=""
                  type="text"
                  value=""
                />
                
      
              </div>
              
      
              <!--?lit$569777623$-->
              
    
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          N {
            "_$AA": <!--?lit$569777623$-->,
            "_$AB": 
      ,
            "_$AH": Symbol(lit-nothing),
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          _t {
            "_$AH": [Function],
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              part="text-field-icon"
            >
              
          
              <slot
                name="textfield-icon"
              />
              
        
            </div>,
            "name": "click",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 5,
          },
          R {
            "_$AH": "text-field-input",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "class",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "id",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "value",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "placeholder",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": false,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "autofocus",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          bt {
            "_$AH": false,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "disabled",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          bt {
            "_$AH": false,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "readonly",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          bt {
            "_$AH": false,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "required",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          bt {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "pattern",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          bt {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "minlength",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          bt {
            "_$AH": "",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "maxlength",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 4,
          },
          _t {
            "_$AH": [Function],
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <input
              autofocus="false"
              class="text-field-input"
              id=""
              part="text-field-input"
              placeholder=""
              type="text"
              value=""
            />,
            "name": "input",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 5,
          },
          N {
            "_$AA": <!--?lit$569777623$-->,
            "_$AB": 
    ,
            "_$AH": Symbol(lit-nothing),
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
